+sw_tests/+unit_tests/unittest_spinw_spinwave.m

```diff
@@ -0,0 +1,471 @@
+        end
+         function test_sw_qh5_saveSabp_incommensurate(testCase)
+             qpts = testCase.qh5;
+             sw_out = testCase.swobj_tri.spinwave(qpts, ...
+                                                  'saveSabp', true);
+             expected_Sabp = zeros(3, 3, 2, 5);
+             expected_Sabp(:, :, :, [1 5]) = repmat( ...
+                 diag([435.71079, 435.71079, 6.45497e-4]), 1, 1, 2, 2);
+             expected_Sabp(:, :, :, [2 4]) = repmat( ...
+                 diag([0.59293, 0.59293, 0.47434]), 1, 1, 2, 2);
+             expected_Sabp(:, :, :, 3) = repmat( ...
+                 diag([0.1875, 0.1875, 1.5]), 1, 1, 2, 1);
+             omegap_vals =  [1.16190e-2 4.74342 3];
+             expected_omegap = [ omegap_vals  flip(omegap_vals(1:2)); ...
+                                -omegap_vals -flip(omegap_vals(1:2))];
+
+             testCase.verify_val(expected_Sabp, sw_out.Sabp, 'rel_tol', 1e-5);
+             testCase.verify_val(expected_omegap, sw_out.omegap, 'rel_tol', 1e-5);
+        end
+        function test_sw_qh5_fitmode(testCase)
+            qpts = testCase.qh5;
+            sw_out = testCase.swobj.spinwave(qpts, 'fitmode', true);
+            % fitmode automatically turns off sortMode
+            expected_sw = testCase.swobj.spinwave(qpts, 'sortMode', false);
+            expected_sw = rmfield(expected_sw, {'obj', 'datestart', 'dateend'});
+            testCase.verify_spinwave(expected_sw, sw_out);
+        end
+        function test_symbolic(testCase)
+            % Test that spinw.spinwavesym() is called if spinw.symbolic==true and spinw.spinwave() is called
+            % Mock the spinwavesym and symbolic methods
+            [mocksw, bh] = testCase.createMock(?spinw, 'MockedMethods', {'symbolic', 'spinwavesym'});
+            % Make sure that spinw thinks it is symbolic
+            testCase.assignOutputsWhen(withAnyInputs(bh.symbolic), true);
+            % Make sure that spinwavesym outputs a sample spectra struct
+            hkl = [1 2; 3 4; 5 6];
+            out_spec = struct('hkl', hkl, 'swConv', [1; 2]);
+            testCase.assignOutputsWhen(withAnyInputs(bh.spinwavesym), out_spec);
+            % Call spinwave
+            spec = mocksw.spinwave(hkl);
+            testCase.assertCalled(withExactInputs(bh.spinwavesym()));
+            testCase.assertEqual(spec, out_spec);
+        end
+        function test_incommensurate(testCase)
+            % Tests that incommensurate calculation is ok
+            hkl = {[0 0 0] [0 1 0] [1 0 0] 5};
+            % Create copy to avoid changing obj for other tests
+            swobj = copy(testCase.swobj);
+            commensurate_spec = swobj.spinwave(hkl);
+            swobj.genmagstr('mode', 'direct', 'k', [0.123 0 0], 'n', [1 0 0], 'S', [0; 1; 0]);
+            % Forcing incomm struct means exchange parameters don't agree, so we need 'hermit' false
+            incomm_spec = swobj.spinwave(hkl, 'hermit', false);
```

Comment:
I have no idea, this test is left over from the initial spinwave unit testing file, and I wonder if this test is telling us anything useful any more?  `test_sw_qh5_saveSabp_incommensurate` tells us an incommensurate calculation of `Sabp` works, so maybe this test should be removed...

Reply:
@mducle what do you think?

Reply:
If `'hermit', true` is given then the calculation will fail with [an error](https://github.com/SpinW/spinw/blob/97b6220f698eec003041767bfe9a4595b0520afc/swfiles/%40spinw/spinwave.m#L820) saying the Hamiltonian is not positive definite / hermitian.

It basically says that the magnetic structure (a cycloid propagating along `[h00]` with moments in the `bc`-plane) does not agree with the input exchange interactions - which is just a nearest neighbour ferromagnetic Heisenberg interaction - `-eye(3)` in this particular case; to stabilise the cycloid you'd need to have anti-symmetric exchange (e.g. a Dzyaloshinskii-Moriya interaction which can be set using `addmatrix` with a 3-vector for the `'value'`) or some further neighbour exchanges. 

This was basically to test that the incommensurate calculations run and results in 3x times the number of modes as the commensurate calculations. Which `test_sw_qh5_saveSabp_incommensurate` doesn't quite do because it uses a different model (the triangular lattice model which really is correctly described by the incommensurate approach unlike this particular example here)...

So, I would still like to keep this test, but maybe also have a check that if `'hermit', true` is given that the "non-positive definite" error is raised... You can also note that the test is to check that the IC calculations give 3 times the number of modes in accordance with equation (40) of Toth and Lake.

Reply:
Have edited this test to use the 2nd neighbour afm chain